cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -63,22 +63,25 @@ func kubeDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interfac
 		return err
 	}
 
-	var dnsBindAddr string
+	var dnsBindAddr, dnsProbeAddr string
 	if dnsip.To4() == nil {
 		dnsBindAddr = "::1"
+		dnsProbeAddr = "[::1]"
 	} else {
 		dnsBindAddr = "127.0.0.1"
+		dnsProbeAddr = "127.0.0.1"
```

Comment:
I'd rather do `dnsProbeAddr = "[" + dnsBindAddr + "]"` and `dnsProbeAddr = dnsBindAddr` instead of repeating constants. Beside that, looks good.

Reply:
Agreed, will change.

Reply:
Instead, could use net.JoinHostPort() and pass in dnsBindAddr and port number, using that host/port in template.

Reply:
@pmichali : There would be a bit of a tradeoff in using net.JoinHostPort(), in that it would require more variables being passed to the template (since the template uses 2 separate IP:port combinations, one for port 53, and one for port 10053). For example, the variable setup would look (roughly) like this:
```
	dnsBindAddr = "::1"
	kubeDNSEndpoint = net.JoinHostPort(dnsBindAddr, "53")
	dnsMasqEndpoint = net.JoinHostPort(dnsBindAddr, "10053")
```
And the instantiation of the template would look like this:
```
	struct{ ImageRepository, Arch, Version, DNSBindAddr, KubeDNSEndpoint, DNSMasqEndpoint, DNSDomain, DNSProbeType, MasterTaintKey string }{
```
with the template definition being:
```
       - --probe=kubedns,{{ .KubeDNSEndPoint }},kubernetes.default.svc.{{ .DNSDomain }},5,{{ .DNSProbeType }}
       - --probe=dnsmasq,{{ .DNSMasqEndpoint }},kubernetes.default.svc.{{ .DNSDomain }},5,{{ .DNSProbeType }}
```
My preference would be the less verbose option adding square brackets directly (with one template variable for DNS probe definitions):
```
    dnsProbeAddr = "[" + dnsBindAddr + "]"
```